src/rank_llm/rerank/pairwise/pairwise_inference_handler.py

```diff
@@ -0,0 +1,83 @@
+from typing import Any, Dict
+
+from transformers import T5Tokenizer
+
+from rank_llm.data import Result
+from rank_llm.rerank.inference_handler import BaseInferenceHandler
+
+
+class PairwiseInferenceHandler(BaseInferenceHandler):
+    def __init__(self, template: Dict[str, str]):
+        super().__init__(template)
+
+    def _validate_template(self, template: Dict[str, str], strict: bool = False):
+        TEMPLATE_SECTIONS = {
+            # Format:
+            # "template_key": {
+            #    "required": True/False,  # Whether the section itself is mandatory
+            #    "required_placeholders": set(),  # Placeholders that must exist in this section
+            #    "allowed_placeholders": set()    # All allowed placeholders (including required ones)
+            # }
+            "body": {
+                "required": True,
+                "required_placeholders": {"query", "doc1", "doc2"},
+                "allowed_placeholders": set(),
+            },
+        }
+
+        # Validate the method value
+        if template["method"] != "pairwise":
+            raise ValueError(
+                f'Incorrect method type, expected "pairwise", got {template["method"]}'
+            )
+
+        self._general_validation(
+            template=template, template_section=TEMPLATE_SECTIONS, strict=strict
+        )
+
+    # TODO (issue #273): May need to add prefix/suffix generation function later
+
+    def _generate_body(
+        self,
+        result: Result,
+        index1: int,
+        index2: int,
+        max_token: int,
+        tokenizer: T5Tokenizer,
+    ) -> str:
+        doc1_raw = self._convert_doc_to_prompt_content(
```

Comment:
I think using max_token as a limit for both docs is an existing mistake, because the total length can end up to be 2 x max_token, please fix